src/ReachSets/discretize.jl

```diff
@@ -800,31 +800,30 @@ function discretize_interval_matrix(𝑆::InitialValueProblem, δ::Float64,
     n = size(A, 1)
     linear_maps = Vector{LinearMap{N}}(undef, order > 2 ? 3 : 2)
 
-    IδW = 1/2 * δ^2 * A + 1/6 * δ^3 * A * A
-    @inbounds for i in 1:n
-        IδW[i, i] *= δ
-    end
+    A² = A * A
+    IδW = δ*I + 1/2 * δ^2 * A + 1/6 * δ^3 * A²
     linear_maps[1] = LinearMap(IδW, U0)
 
-    linear_maps[2] = LinearMap(_expm_remainder(A, δ, order; n=n) * δ, U0)
+    E = _expm_remainder(A, δ, order; n=n)
+    linear_maps[2] = LinearMap(E*δ, U0)
 
     zero_interval = IntervalMatrices.Interval(zero(N), zero(N))
     if order > 2
-        fac_i_plus_1 = 6
-        Ai = A * A
-        δ_iplus1 = δ^3
+        # i = 2
+        αᵢ₊₁ = 6 # factorial of (i+1)
+        Aⁱ = A²
+        δⁱ⁺¹ = δ^3
         M_sum = IntervalMatrix(fill(zero_interval, size(A)))
         @inbounds for i in 3:order
-            fac_i_plus_1 *= i+1
-            δ_iplus1 *= δ
-            Ai *= A
-            1/fac_i_plus_1 * δ_iplus1 * Ai
+            αᵢ₊₁ *= i+1
+            δⁱ⁺¹ *= δ
+            Aⁱ *= A
+            M_sum += 1/αᵢ₊₁ * Aⁱ * δⁱ⁺¹
```

Comment:
Before I first did the scalar computation and then the matrix scaling. Now your code needs to do two matrix operations. I suspect that the previous version was faster. Was it wrong?

Reply:
Indeed, i missed that point. Other computations should dominate, but i agree it is better to add parens over the scalar terms.